torchinductor/lowering.py

```diff
@@ -1296,32 +1296,76 @@ def fn(idx):
     )
 
 
-@register_lowering(aten.scatter_add_, type_promote=False)
-def scatter_add_(self, dim: int, index, src):
+@register_lowering(aten.scatter_, type_promote=False)
+def scatter_(self, dim: int, index, src, *, reduce: str = None):
+    if reduce == "add":
+        reduce = "sum"
+    elif reduce == "multiply":
+        reduce = "prod"
+    else:
+        assert reduce is None
+    return scatter_reduce_(self, dim, index, src, reduce)
+
+
+@register_lowering(aten.scatter_reduce_, type_promote=False)
+def scatter_reduce_(self, dim: int, index, src, reduce, *, include_self: bool = True):
+    # TODO: Need to support more reduction type
+    assert reduce is None or reduce in {"sum"}
     assert isinstance(self, TensorBox)
     assert "int" in str(index.get_dtype())
     assert 0 <= dim < len(self.get_size())
-    src = to_dtype(src, self.get_dtype())
+
     self.realize()
     V.graph.realize_users_of(self.get_name())
-
     index_loader = index.make_loader()
 
     def output_indexer(idx):
         indirect_idx = list(idx)
         indirect_idx[dim] = ops.indirect_indexing(index_loader(idx))
         return indirect_idx
 
+    def fn(idx):
+        if isinstance(src, TensorBox):
+            src_loader = src.make_loader()
```

Comment:
We should call make_loader() outside of fn in order to handle mutation properly.  (make_loader will resolve where StorageBox.data points at the time it is called.  Mutation swings the StorageBox.data pointer to the new value.)